master/pkg/model/user.go

```diff
@@ -13,8 +13,8 @@ import (
 )
 
 var (
-	// EmptyPassword is the empty password (i.e., the empty string).
-	EmptyPassword = null.NewString("", false)
+	// DefaultPassword is a default password set in master config.
+	DefaultPassword = null.NewString("", false)
```

Comment:
On Kubernetes the behavior of DefaultPassword is somewhat misleading

New users created still get a "blank" password just ```admin``` and ```determined``` (the two users who are created initially)

The name ```defaultPassword``` might not be the best to describe this? The poor naming comes from the Helm chart so maybe we can make the master config one more accurate something like ```admin_determined_initial_password```? We can still keep the Helm chart the same to avoid breaking existing users.

Reply:
There are maybe two ways this could be implemented: 

1. on start up we search for any users without a password and set it to the desired value in the database
2. or, when we validate against a user with an invalid hash we use the default instead

I was advocating for approach #2 in which case I think the name is appropriate.

Ideally tho I think any empty empty passwords should be assigned a random value and recorded to disk (eg as initial_passwords.yaml), but that is a fairly big departure from our current behavior.

Reply:
This isn't how the current Helm option ```defaultPassword``` works though. It only runs on first install of the cluster and only will apply to the ```determined``` and ```admin``` user. We could potentially break peoples clusters and workflows doing it this way.

The most correct to match the behavior would need to check it is a first run either (probably could with cluster ID being unset on startup).  

Maybe we should even mark this as internal with an underscore prefix and not document it since this feature is meant to reduce the maintenance burden from the k8s method of doing it. I'm not too concerned about this behaviour being somewhat suboptiomal since we haven't gotten a user ask to change it as far as I understand. Any sufficiently large users should be using an external IDP.